File: test/javascript/healthCheck.js
var assert = require('assert');
var NodeSwaggerAem = require('node-swagger-aem');

describe('healthCheck', function() {

  describe('getAemHealthCheck', function() {
    it('should respond without error', function(done) {
      var defaultClient = NodeSwaggerAem.ApiClient.instance;
      var aemAuth = defaultClient.authentications['aemAuth'];
      aemAuth.username = 'admin';
      aemAuth.password = 'admin';

      var api = new NodeSwaggerAem.CustomApi();

      // Stub the underlying HTTP call so no real AEM server is needed
      var calledWith = null;
      api.apiClient.callApi = function() {
        calledWith = { path: arguments[0], httpMethod: arguments[1] };
        var callback = arguments[arguments.length - 1];
        callback(null, 'OK', { status: 200 });
      };

      var callback = function(error, data, response) {
        if (error) {
          console.error(error);
        } else {
          console.log('API called successfully. Returned data: ' + data);
          assert.equal(data, 'OK');
          assert.equal(calledWith.path, '/system/health');
          assert.equal(calledWith.httpMethod, 'GET');
        }
        done(error);
      };
      api.getAemHealthCheck({}, callback);
    });
  });

});

File: test/python/health_check.py
import unittest
from unittest.mock import patch
import urllib3
import swaggeraem
from swaggeraem.api import custom_api

class TestHealthCheck(unittest.TestCase):

    @patch.object(urllib3.PoolManager, 'request')
    def test_get_aem_health_check(self, mock_request):

      # Stub the underlying HTTP call so no real AEM server is needed
      mock_request.return_value = urllib3.HTTPResponse(
          body=b'"OK"',
          status=200,
          headers={'Content-Type': 'application/json'}
      )

      configuration = swaggeraem.Configuration(
          host = "http://localhost:4502",
          username = "admin",
          password = "admin"
      )

      with swaggeraem.ApiClient(configuration) as api_client:

          api_instance = custom_api.CustomApi(api_client)

          try:
              api_response = api_instance.get_aem_health_check()
              print("The response of CustomApi->get_aem_health_check:\n")
              print(api_response)
              assert api_response == 'OK'
              method, url = mock_request.call_args[0]
              assert method == 'GET'
              assert url.endswith('/system/health')
          except swaggeraem.ApiException as e:
              self.fail('Exception when calling CustomApi->get_aem_health_check: %s\n' % e)
